augur/argparse_.py

```diff
@@ -18,6 +19,14 @@
 SKIP_AUTO_DEFAULT_IN_HELP = "%(default).0s"
 
 
+# We use nargs='+' with action='extend' for a more intuitive user experience,
+# but it has the side effect of extending instead of replacing a default value
+# defined in add_argument. Allow users to override the default value by first
+# defaulting to this constant, then setting the intended default value at a
+# later point in the code.
+DEFER_LIST_DEFAULT: List[Any] = []
```

Comment:
_non-blocking_ 

Thoughts on writing a custom Argparse Action that handles the default for extend the expected way? We could bring back `ExtendAction` dropped in https://github.com/nextstrain/augur/commit/f691899e737ec3b5b9037487cf5ae3c9acd06e9d

Reply:
That sounds much better, I'll give it a shot in this PR.

Reply:
Prior art here:

https://github.com/nextstrain/cli/blob/9bf646b0c795d04658a6f6807d74428b7c173995/nextstrain/cli/argparse.py#L183-L197

Reply:
Switched to `ExtendOverwriteDefault` (inspired by Nextstrain CLI's `AppendOverwriteDefault`) in [force-push](https://github.com/nextstrain/augur/compare/20206183a811c37e6d22fc3da9d57113fc8ee186..e1fbbfc94a21505f5eedcaaf9073cd181c4dbea8).